docs/site/Loopback-component-authorization.md

```diff
@@ -8,543 +8,377 @@ permalink: /doc/en/lb4/Loopback-component-authorization.html
+> Usually the `authorize` functions are bound through a provider as below
 
 ```ts
-import {Component, ProviderMap} from '@loopback/core';
-import {AuthorizationBindings} from './keys';
-import {AuthorizeActionProvider} from './providers/authorization-action.provider';
-import {AuthorizationMetadataProvider} from './providers/authorization-metadata.provider';
-import {UserPermissionsProvider} from './providers/user-permissions.provider';
-
-export class AuthorizationComponent implements Component {
-  providers?: ProviderMap;
-
-  constructor() {
-    this.providers = {
-      [AuthorizationBindings.AUTHORIZE_ACTION.key]: AuthorizeActionProvider,
-      [AuthorizationBindings.METADATA.key]: AuthorizationMetadataProvider,
-      [AuthorizationBindings.USER_PERMISSIONS.key]: UserPermissionsProvider,
-    };
+class MyAuthorizationProvider implements Provider<Authorizer> {
+  /**
+   * @returns an authenticating function
+   *
+   */
+  value(): Authorizer {
+    return this.authorize.bind(this);
   }
-}
-```
 
-You can see that we have used the same binding keys which we created earlier.
-
-Now, its time to create our method decorator function. Here it is. We will be
-using the same metadata accessor key which we created earlier and the metadata
-interface for accessing the data in decorator.
-
-{% include code-caption.html content="/src/authorization/decorators/authorize.decorator.ts" %}
-
-```ts
-import {MethodDecoratorFactory} from '@loopback/core';
-import {AuthorizationMetadata} from '../types';
-import {AUTHORIZATION_METADATA_ACCESSOR} from '../keys';
-
-export function authorize(permissions: string[]) {
-  return MethodDecoratorFactory.createDecorator<AuthorizationMetadata>(
-    AUTHORIZATION_METADATA_ACCESSOR,
-    {
-      permissions: permissions || [],
-    },
-  );
-}
-```
-
-For error handling keys, lets create an enum.
-
-{% include code-caption.html content="/src/authorization/error-keys.ts" %}
-
-```ts
-export const enum AuthorizeErrorKeys {
-  NotAllowedAccess = 'Not Allowed Access',
+  async authorize(
+    context: AuthorizationContext,
+    metadata: AuthorizationMetadata,
+  ) {
+    events.push(context.resource);
+    if (
+      context.resource === 'OrderController.prototype.cancelOrder' &&
+      context.principals[0].name === 'user-01'
+    ) {
+      return AuthorizationDecision.DENY;
+    }
+    return AuthorizationDecision.ALLOW;
+  }
 }
 ```
 
-Finally, we put everything together in one index file.
+The `AuthorizationContext` parameter of the `authorize` function contains the
+current principal (in the example given above,that would be the current user
+invoking `cancelOrder`) and details of the invoked endpoint. The
+`AuthorizationMetadata` parameter of the `authorize` function contains all the
+details provided in the invoked method's decorator.
 
-{% include code-caption.html content="/src/authorization/index.ts" %}
+> and then tagged to an application as `AuthorizationTags.AUTHORIZER` as below.
```

Comment:
what sentence is `and then tagged` belonging to? And why is it using '>' which indents it?

![image](https://user-images.githubusercontent.com/6864736/70177145-d3623680-16a7-11ea-8513-c4d47591d5a2.png)

Reply:
I thought highlighting this way , gives better readability.

Reply:
any better suggestions welcomed

Reply:
I have removed the `and` conjunction now.